mne/epochs.py

```diff
@@ -809,14 +809,22 @@ def __next__(self, *args, **kwargs):
         """Provide a wrapper for Py3k."""
         return self.next(*args, **kwargs)
 
-    def average(self, picks=None):
-        """Compute average of epochs.
+    def average(self, picks=None, method="mean"):
+        """Compute an aggregate over epochs. Defaults to the mean.
 
         Parameters
         ----------
         picks : array-like of int | None
             If None only MEG, EEG, SEEG, ECoG, and fNIRS channels are kept
             otherwise the channels indices in picks are kept.
+        method : str | callable
+            How to combine the data. If "mean", "median" or "std", the mean,
+            median or standard deviation are returned.
+            Otherwise, must be a callable which, when passed an array of shape
+            (n_epochs, n_channels, n_time) returns an array of shape
+            (n_channels, n_time).
+
+            e.g.::    epochs.average(method=lambda x: np.median(x, axis=0))
```

Comment:
This will not render properly, and it's getting a bit verbose. Let's put this in an `Examples` section, like:
```
...::

>>> from scipy.stats import trim_mean  # doctest:+SKIP
>>> epochs.average(method=lambda x: trim_mean(x, 10, axis=0))  # doctest:+SKIP
```